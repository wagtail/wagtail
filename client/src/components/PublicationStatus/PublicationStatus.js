import React from 'react';

/**
 * Displays the publication status of a page in a pill.
 */
const PublicationStatus = ({ status }) => (
  <span className={`o-pill c-status${status.live ? ' c-status--live' : ''}`}>
    {status.status}
  </span>
);

PublicationStatus.propTypes = {
  status: React.PropTypes.shape({
    live: React.PropTypes.bool.isRequired,
    status: React.PropTypes.string.isRequired,
  }).isRequired,
};

export default PublicationStatus;
