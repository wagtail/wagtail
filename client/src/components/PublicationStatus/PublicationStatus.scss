.c-status {
  @apply w-bg-black-50 w-tracking-tight w-text-text-label-menus-default;
  display: inline-block;
  padding: 0.2em 0.5em;
  border-radius: 0.25em;
  vertical-align: middle;
  line-height: 1.5;
  // stylelint-disable-next-line property-disallowed-list
  text-transform: uppercase;
  font-size: 0.625rem;
}
