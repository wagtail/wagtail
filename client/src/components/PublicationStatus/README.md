# PublicationStatus

Displays the publication status of a page in a pill.

## Usage

```javascript
import { PublicationStatus } from 'wagtail';

render(
    <PublicationStatus
        status={status}
    />
);
```

### Available props

- `status`: status object coming from the admin API. If no status is given, component renders to null.
