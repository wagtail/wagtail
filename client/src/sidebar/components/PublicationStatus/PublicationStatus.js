import PropTypes from 'prop-types';
import React from 'react';

/**
 * Displays the publication status of a page in a pill.
 */
const PublicationStatus = ({ status }) => (
  <span className={`o-pill c-status${status.live ? ' c-status--live' : ''}`}>
    {status.status}
  </span>
);

PublicationStatus.propTypes = {
  status: PropTypes.shape({
    live: PropTypes.bool.isRequired,
    status: PropTypes.string.isRequired,
  }).isRequired,
};

export default PublicationStatus;
