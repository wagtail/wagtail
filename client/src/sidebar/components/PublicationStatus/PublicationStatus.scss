.c-status {
  @apply w-bg-black/50 w-tracking-tight w-text-white/85;
  // stylelint-disable-next-line property-disallowed-list
  text-transform: uppercase;
  font-size: 10px;
}
