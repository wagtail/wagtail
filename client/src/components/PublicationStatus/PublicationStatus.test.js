import React from 'react';
import { shallow } from 'enzyme';

import PublicationStatus from './PublicationStatus';

describe('PublicationStatus', () => {
  it('exists', () => {
    expect(PublicationStatus).toBeDefined();
  });

  it('#status live', () => {
    expect(shallow((
      <PublicationStatus
        status={{
          status: 'live + draft',
          live: true,
          has_unpublished_changes: true,
        }}
      />
    ))).toMatchSnapshot();
  });

  it('#status not live', () => {
    expect(shallow((
      <PublicationStatus
        status={{
          status: 'live + draft',
          live: false,
          has_unpublished_changes: true,
        }}
      />
    ))).toMatchSnapshot();
  });
});
