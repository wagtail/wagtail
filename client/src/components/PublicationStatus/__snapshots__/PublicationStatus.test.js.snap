// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicationStatus #status live 1`] = `
<span
  className="c-status c-status--live"
>
  live + draft
</span>
`;

exports[`PublicationStatus #status not live 1`] = `
<span
  className="c-status"
>
  live + draft
</span>
`;
